src/wg/color/rgb/format.d

```diff
@@ -42,14 +52,324 @@ struct RGBFormatDescriptor
 
     byte bits;
     byte alignment;
-    bool bigEndian;
-    bool hasSharedExponent;
-
-    ushort componentsPresent;
+    ushort flags;
 
-    const(ComponentDesc)[] components;
+    ComponentDesc[] components;
 
     const(RGBColorSpace)* colorSpace;
 
-    string userData;
+    const(char)[] userData;
```

Comment:
I kinda presumed it was text (since it's stuck on the end of the format string)... but maybe? dunno. I could remove it altogether...?

Reply:
Ah, yes, you're right, `userData` is text

Reply:
I just thought it could be useful for user tag type things... but i could also remove it.

I was inspired by  the thought that some games consoles have weird texture formats with some odd rules, and there's no point codifying them in the spec, but a user could put a tag in there that would allow them to detect and implement the format details themselves.